index.ts

```diff
@@ -0,0 +1,476 @@
+//	  - Descriptions are UI only and can be large so aren't included in the Dex by
+//	    default, but they may be loaded in bulk asynchronously for the desired
+//	    languages.
+//   - The aliases.json data is used internally (to implement get/getByID etc) but
+//     not exported.
+//   - `forGenSync` methods can be added everywhere `forGen` exists as a non-Web
+//     convenience feature, though simply using `await` should be preferred.
+//
+//  *** TODO ***:
+//
+//   - 'Species' name is problematic as it is plural
+//   - How should 'letsgo'/'stadium' (mods) be handled? Option: use `forMod`
+//		 instead of `forGen`, and add a `genToMod(gen: Generation)` helper method
+//		 (plus overloads?)
+//	  - Instead of implemented `next()` directly, should `keys/values/entries` style
+//	    Iterator methods be written?
+//
+
+// data.ts
+
+type ID = '' | string & {__isID: true};
+toID(s: string): ID;
+
+type Generation = 1 | 2 | 3 | 4 | 5 | 6 | 7;
+
+type DataKind = 'Ability' | 'Move' | 'Item' | 'Species';
+type Nonstandard = 'Glitch' | 'Past' | 'Future' | 'CAP' | 'LGPE' | 'Pokestar' | 'Custom';
+interface Data {
+	id: ID;
+	kind: DataKind;
+	name: string;
+	num: number;
+	gen: Generation;
+	isUnreleased?: boolean;
+	isNonstandard: Nonstandard | null;
+}
+
+// natures.ts
+
+type NatureName = 'Adamant' | 'Bashful' | 'Bold' | 'Brave' | 'Calm' | 'Careful' | 'Docile' | 'Gentle' |
+	'Hardy' | 'Hasty' | 'Impish' | 'Jolly' | 'Lax' | 'Lonely' | 'Mild' | 'Modest' | 'Naive' | 'Naughty' |
+	'Quiet' | 'Quirky' | 'Rash' | 'Relaxed' | 'Sassy' | 'Serious' | 'Timid';
+
+class Natures {
+	get(name: string): Nature | undefined;
+	next(): Nature | true;
+}
+
+interface Nature {
+	id: ID;
+	name: NatureName;
+	plus?: StatName;
+	minus?: StatName;
+}
+
+// abilities.ts
+
+interface Ability extends Data {
+	kind: 'Ability';
+	suppressWeather: boolean;
+	isUnbreakable?: boolean
+}
+
+class Abilities {
+	static forGen(gen: Generation): Promise<Abilities>;
+
+	get(name: string): Ability | undefined;
+	getByID(id: ID): Ability | undefined;
+	next(): Ability | true;
+}
+
+// items.ts
+
+interface Item extends Data {
+	kind: 'Item';
+
+	fling?: { basePower: number status?: string volatileStatus?: string };
+	forcedForme?: string;
+	ignoreKlutz?: boolean;
+	infiltrates?: boolean;
+	isBerry?: boolean;
+	isChoice?: boolean;
+	isGem?: boolean;
+	isPokeball?: boolean;
+	megaEvolves?: string;
+	megaStone?: string;
+	naturalGift?: {basePower: number, type: TypeName};
+	onDrive?: TypeName;
+	onMemory?: TypeName;
+	onPlate?: TypeName;
+	recoil?: [number, number];
+	status?: string;
+	weather?: string;
+	zMove?: string | true | null;
+	zMoveFrom?: string;
+	zMoveType?: TypeName;
+	zMoveUser?: string[] | null;
+}
+
+class Items {
+	static forGen(gen: Generation): Promise<Items>;
+
+	get(name: string): Item | undefined;
+	getByID(id: ID): Item | undefined;
+	next(): Item | true;
+}
+
+// moves.ts
+
+type MoveFlag = 1 | 0;
```

Comment:
isn't this just `boolean`?

Reply:
@Marty - was this coded this way just to save on typing?

Reply:
This is basically legacy, but I remember asking about it before implementing it. @Zarel recommended using an object for `move.flags` like everything else implemented in moves.js at the time, instead of an array. Since then, other stuff has been switched to an array but `flags` has remained like this.

Reply:
I'm fine with `flags` being an array.